### PR TITLE
Op eval tests

### LINKED ISSUE #846: Improve Unit Test for operational eval to 20%

### PR DESCRIPTION
**Pull request by jhsinger-klotho**
closes #846 

adding unit tests to op eval, causing some refactor for interfaces to make code unit testable

### Standard checks

- **Unit tests**: Any special considerations?
- **Docs**: Do we need to update any docs, internal or public?
- **Backwards compatibility**: Will this break existing apps? If so, what would be the extra work required to keep them working?

### CODE REVIEW

File: pkg/engine2/operational_eval/vertex_property_test.go [hunk 1]
```diff
@@ -230,3 +231,136 @@ func Test_propertyVertex_Dependencies(t *testing.T) {
...
+		mocks   func(mockProperty *MockProperty, mockRc *MockResourceConfigurer, data testData)
+		wantErr bool
+	}
+	tests := []testData{
+		{
+			name: "existing value and no constraints",
+			ref: construct.PropertyRef{
+				Resource: id,
+				Property: "test",
+			},
```
Comment (gordon-klotho) [inline pkg/engine2/operational_eval/vertex_property_test.go:251]: nit: could use `graphtest.ParseRef` to let these be strings to be more compact without losing readability

File: pkg/engine2/testdata/vpc_import_to_lambda.dataflow-viz.yaml [hunk 2]
```diff
@@ -4,6 +4,7 @@ resources:
     children:
         - aws:security_group:vpc:lambda_function-security_group
         - aws:subnet:vpc:subnet1
+        - aws:subnet:vpc:subnet2
```
Comment (gordon-klotho) [inline pkg/engine2/testdata/vpc_import_to_lambda.dataflow-viz.yaml:7]: I assume this changed due to the addition of the namespace (shown in the `.expect.yaml`)

File: pkg/engine2/testdata/vpc_import_to_lambda.expect.yaml [hunk 3]
```diff
@@ -31,7 +31,7 @@ resources:
             - aws:security_group:vpc:lambda_function-security_group
         Subnets:
             - aws:subnet:vpc:subnet1
-            - aws:subnet:subnet2
+            - aws:subnet:vpc:subnet2
```
Comment (gordon-klotho) [inline pkg/engine2/testdata/vpc_import_to_lambda.expect.yaml:34]: Was this fixed by the `&& setConstraint.Operator == ""` to the vertex property?

File: pkg/knowledge_base2/kb.go [hunk 4]
```diff
@@ -12,6 +12,7 @@ import (
 )
 
 //go:generate mockgen --source=./kb.go -destination=./template_kb_mock_test.go -package=knowledgebase2
+//go:generate mockgen --source=./kb.go -destination=../engine2/operational_eval/template_kb_mock_test.go -package=operational_eval
```
Comment (gordon-klotho) [inline pkg/knowledge_base2/kb.go:15]: This is really awkward - should just generate one mock. That's why there's the `graphtest`, `enginetest`, `kbtesting` etc packages. It's also a little awkward, but it's standard for Go for reusable test utilities (see for eg. [httptest](https://pkg.go.dev/net/http/httptest) in the stdlib).

**Review by gordon-klotho — APPROVED** [review]
Nothing blocking

File: pkg/knowledge_base2/kb.go (hunk 4 above)
Comment (jhsinger-klotho) [reply pkg/knowledge_base2/kb.go:15]: well get into the circular dependencies again. Theres no way to test the actual solution context package for example. but we could try a different method, i just think this gets rid of that worry

File: pkg/engine2/testdata/vpc_import_to_lambda.expect.yaml (hunk 3 above)
Comment (jhsinger-klotho) [reply pkg/engine2/testdata/vpc_import_to_lambda.expect.yaml:34]: yup

**Pull request merged**